src/uu/expr/src/syntax_tree.rs

```diff
@@ -173,6 +170,87 @@ impl StringOp {
     }
 }
 
+/// Check errors with a supplied regular expression
+///
+/// GNU coreutils shows messages for invalid regular expressions
+/// differently from the oniguruma library used by the regex crate.
+/// This method attempts to do these checks manually in one pass
+/// through the regular expression.
+fn validate_regex(pattern: &str) -> ExprResult<()> {
```

Comment:
there isn't any crate doing it already ?

Reply:
I don't think so? Regex (and related) crates don't use posix syntax. tests/expr/expr.pl does check that `\{` and `{` are treated opposite of how they would be in the regex

Reply:
maybe rename this function to make it more specific ?
like
validate_regex_posix?